File: Fibonacci/C++/Fibonacci.cpp
/*              LA SUCESIÓN DE FIBONACCI
 * Escribe un programa que imprima los 50 primeros números de la sucesión
 * de Fibonacci empezando en 0.
 * - La serie Fibonacci se compone por una sucesión de números en
 *   la que el siguiente siempre es la suma de los dos anteriores.
 *   0, 1, 1, 2, 3, 5, 8, 13...
 */ 
//Include
#include <iostream>

//Usos
using namespace std;

int main(int argc, char const *argv[])
{
    //Inicializamos las variables
    int num = 50;
    long primero = 0;
    long segundo = 1;

    //Mostramos por pantalla al usuario lo que vamos a imprimir
    cout << "Los primeros " << num << " números de la sucesión de Fibonacci: " << endl;
    for (int i = 0; i < num; i++)
    {
        cout << primero << " ";

        //Calculamos los siguientes numeros sumando los dos anteriores
        long siguiente = primero + segundo;

        //Actualizamos los terminos para la siguiente vuelta del bucle
        primero = segundo;
        segundo = siguiente;
    }

    return 0;
}

File: Fibonacci/JavaScript/Fibonacci.js
//Inicializamos las variables
let num = 50;
let primero = 0;
let segundo = 1;

//Avisamos al usuario de que vamos a imprimir los 50 primeros numeros de la secuencia Fibonacci
console.log("Los primeros " + num + " de la secuencia de Fibonacci son: ");

for (let index = 0; index < num; index++) {
    //Mostramos el primer numero
    console.log(primero + " ");
    
    //Calculamos el siguiente termino sumando los dos anteriores
    let siguiente = primero + segundo;

    //Ajustamos el bucle para las siguientes iteraciones
    primero = segundo;
    segundo = siguiente;
    
}